### PR TITLE
Implement lexical binding 'let'

### PR DESCRIPTION
**Pull request by foriequal0**
```SCHEME
(let ((id valExpr)...) body...)
``` 
can be converted into
```SCHEME
((lambda (id ...) body...) valExpr...)
```
as described in both MIT Scheme reference 2.2 Lexical Binding, and SICP section 1.3.2.

### CODE REVIEW

File: src/Scheme/Evaluator.hs [hunk 1]
```diff
@@ -49,12 +48,27 @@ eval (List (Atom "lambda" : varargs@(Atom _) : body)) =
     makeVarargs varargs [] body
 eval (List [Atom "load", String filename]) =
     (liftEnv $ load filename) >>= fmap last . mapM eval
+eval (List (Atom "let" : List bindings : body)) = do
```
Comment (kseo) [inline src/Scheme/Evaluator.hs]: This is more complicated than I expected. Think of it as a macro expansion. Perform the translation in the AST level first and calls `eval` on the output.

Comment (foriequal0) [inline src/Scheme/Evaluator.hs]: makeNormalFunc가 아니라
```haskell
argVals <- mapM eval args
let lambdafied = List $ (List $ (Atom "lambda") : params : body) : argVals)
eval lambdafied
```
이런 느낌 말씀하시는건가요? 처음에 그렇게 했다가 eval을 다시 호출하면
```haskell
 eval (List (Atom "lambda" : List params : body)) = 
```
에서 결국 다시 패턴 매칭 하길래 makeNormalFunc를 직접 사용한거였거든요.

Comment (kseo) [inline src/Scheme/Evaluator.hs]: Yeah, it is much simpler. I would like to make a distinction between desugaring and evaluation. Translating `let` to `lambda` belongs to desugaring. Later we will move this code to `desugar` function which is a separate step before evaluation.

**Comment by kseo** [comment]
This PR is obsoleted by #18

**Pull request closed without merging**